mllib/src/test/scala/org/apache/spark/ml/util/MLTestingUtils.scala

```diff
@@ -268,4 +269,20 @@ object MLTestingUtils extends SparkFunSuite {
     assert(newDatasetF.schema(featuresColName).dataType.equals(new ArrayType(FloatType, false)))
     (newDataset, newDatasetD, newDatasetF)
   }
+
+  def modelPredictionEquals[M <: PredictionModel[_, M]](
```

Comment:
This is asserting that some subset are within tolerances... is it necessary to loosen it this much? I haven't read carefully; why might some be out of tolerances and that's OK? what about a looser tolerance instead?

Reply:
I'm still wondering about this one FWIW

Reply:
yes, I already made the tolerance looser in latest commit for the regressor (maybe you are not looking at latest code?).  Note for classifier there is no tolerance. See update:
```
      MLTestingUtils.testArbitrarilyScaledWeights[DecisionTreeRegressionModel, 
        DecisionTreeRegressor](df.as[LabeledPoint], estimator, 
        MLTestingUtils.modelPredictionEquals(df, _ ~= _ relTol 0.1, 0.99)) 
      MLTestingUtils.testOutliersWithSmallWeights[DecisionTreeRegressionModel, 
        DecisionTreeRegressor](df.as[LabeledPoint], estimator, numClasses, 
        MLTestingUtils.modelPredictionEquals(df, _ ~= _ relTol 0.1, 0.99), 
        outlierRatio = 2) 
      MLTestingUtils.testOversamplingVsWeighting[DecisionTreeRegressionModel, 
        DecisionTreeRegressor](df.as[LabeledPoint], estimator, 
        MLTestingUtils.modelPredictionEquals(df, _ ~= _ relTol 0.01, 1.0), seed)
```

Reply:
I'm wondering about using this method at all. If the problem is that some values might vary moderately from the expected value, can you just set a wide tolerance? why does this method need to merely assert that some values are in tolerance and let some be out of tolerance?

Reply:
Due to propagation of error during model training some predictions may be significantly out of tolerance (specifically for this test that one prediction which varied between values of 0.4 and 4 for the regressor, which is why in the latest update I check for 99 percent of values being within tolerance - I can actually make this bound tighter at 99.9 percent).  If we make the tolerance that large so that the 0.4 vs 4 prediction passes then this test will always pass even if a bug was introduced and suddenly most prediction differences between the models may start to range from 0.4 to 4 which would not be correct, so making the tolerance very large doesn't make sense.  We want to make sure that the models are roughly equivalent but they can't be exactly equal.  At the very least, we want to make sure they don't diverge more with future changes (or if they do it should be for a good reason).

Reply:
OK, let's leave it.